tests/host_reference/staggered_dslash_reference.cpp

```diff
@@ -169,24 +171,21 @@ void stag_mat(ColorSpinorField &out, const GaugeField &fat_link, const GaugeFiel
   // {{m, -D_eo},{-D_oe,m}}, while the CPU verify function does not
   // have the minus sign. Inverting the expected dagger convention
   // solves this discrepancy.
-  stag_dslash(out.Even(), fat_link, long_link, in.Odd(), QUDA_EVEN_PARITY, 1 - daggerBit, dslash_type);
-  stag_dslash(out.Odd(), fat_link, long_link, in.Even(), QUDA_ODD_PARITY, 1 - daggerBit, dslash_type);
+  stag_dslash(out.Even(), fat_link, long_link, in.Odd(), QUDA_EVEN_PARITY, 1 - daggerBit, dslash_type, laplace3D);
+  stag_dslash(out.Odd(), fat_link, long_link, in.Even(), QUDA_ODD_PARITY, 1 - daggerBit, dslash_type, laplace3D);
 
   if (dslash_type == QUDA_LAPLACE_DSLASH) {
-    double kappa = 1.0 / (8 + mass);
+    double kappa = 1.0 / ((laplace3D == 3 ? 6 : 8) + mass);
```

Comment:
iirc `laplace3D` is an integer equal to the dimension being partitioned, so in the off-chance someone is partitioning the `z` direction instead of `t` direction, `laplace3D` would be `2`, and this would be wrong.

Reply:
196cef438